setup.py

```diff
@@ -89,15 +91,34 @@ def run(self):
         plat = sysconfig.get_platform()
         fastjetdir = pathlib.Path(f"build/lib.{plat}-{version}/fastjet")
 
-        shutil.copytree(OUTPUT, fastjetdir / "_fastjet_core")
-
-        for pythondir in (fastjetdir / "_fastjet_core/lib").glob("python*"):
-            sitepackages = pythondir / "site-packages"
-            (sitepackages / "fastjet.py").rename(fastjetdir / "_swig.py")
-            for sharedobj in sitepackages.glob("*.so*"):
-                sharedobj.rename(fastjetdir / sharedobj.parts[-1])
-
-            shutil.rmtree(pythondir)
+        shutil.copytree(OUTPUT, fastjetdir / "_fastjet_core", symlinks=True)
+
+        pythondir = pathlib.Path(
+            subprocess.check_output(
+                """make -f pyinterface/Makefile --eval='print-pythondir:
+\t@echo $(pythondir)
+' print-pythondir""",
+                shell=True,
+                cwd=FASTJET,
+            )
+            .decode()
+            .strip()
+        )
+
+        pyexecdir = pathlib.Path(
+            subprocess.check_output(
+                """make -f pyinterface/Makefile --eval='print-pyexecdir:
+\t@echo $(pyexecdir)
+' print-pyexecdir""",
+                shell=True,
+                cwd=FASTJET,
+            )
+            .decode()
+            .strip()
+        )
```

Comment:
```suggestion
                cwd=FASTJET,
                universal_newlines=True,
            )
            .strip()
        )
```

You might be able to remove the strip here too, not sure. Python 3.7+ calls this "text=True", but we support 3.6. Also, the recommendation is to use `subprocess.run` for Python 3.5+, but not a big deal here.

(I'd be just fine to merge this in then make improvements like the one above - I'll try to work on the macOS support soon, too)